ufl/finiteelement/mixedelement.py

```diff
@@ -56,9 +56,12 @@ def __init__(self, *elements, **kwargs):
 
         # Check that all elements use the same quadrature scheme TODO:
         # We can allow the scheme not to be defined.
-        quad_scheme = elements[0].quadrature_scheme()
-        if not all(e.quadrature_scheme() == quad_scheme for e in elements):
-            error("Quadrature scheme mismatch for sub elements of mixed element.")
+        if len(elements) == 0:
+            quad_scheme = None
+        else:
+            quad_scheme = elements[0].quadrature_scheme()
+            if not all(e.quadrature_scheme() == quad_scheme for e in elements):
+                error("Quadrature scheme mismatch for sub elements of mixed element.")
```

Comment:
Having to do this suggests that one should revisit the design decision of making `VectorElement` and `TensorElement` subclasses of `MixedElement`. Although I don't suggest we do so here, can you open an issue proposing that we split them out? I would advocate keeping `MixedElement` only for a "bag of unrelated elements" and create new `Vector/TensorElement` classes that rank-augment an existing `FiniteElement` object. One could then go one step further and just make `VectorElement` a functional constructor for a `TensorElement`.

Reply:
See #33